components/AppServices/CommunityToolkit.AppServices.SourceGenerators/Diagnostics/Analyzers/InvalidValueSetSerializerUseAnalyzer.cs

```diff
@@ -0,0 +1,81 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using CommunityToolkit.AppServices.SourceGenerators.Extensions;
+using static CommunityToolkit.AppServices.SourceGenerators.Diagnostics.DiagnosticDescriptors;
+
+namespace CommunityToolkit.AppServices.SourceGenerators;
+
+/// <summary>
+/// A diagnostic analyzer that emits diagnostics whenever a value set serializer type is not valid.
+/// </summary>
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+public sealed class InvalidValueSetSerializerUseAnalyzer : DiagnosticAnalyzer
+{
+    /// <inheritdoc/>
+    public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(InvalidValueSetSerializerType, InvalidValueSetSerializerLocation);
+
+    /// <inheritdoc/>
+    public override void Initialize(AnalysisContext context)
+    {
+        context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
+        context.EnableConcurrentExecution();
+
+        // Register a callback for all named type symbols (ie. user defined types)
+        context.RegisterSyntaxNodeAction(static context =>
```

Comment:
Why syntax node action?

Semantic model calls might be more expensive than using symbol/operation action (maybe register operation action for IAttributeOperation?)

Reply:
The analyzer is currently targeting Roslyn 4.3, which doesn't support attribute operations.
We might want to update this in a separate PR, assuming it was actually worth the break 🤔